### PR TITLE
More script/build improvements

### PR DESCRIPTION
**Pull request by Margen67**
See commit history.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+      - '.gitignore'
+      - '.vscode/**'
+  pull_request:
+    paths-ignore:
+      - '.github/*_TEMPLATE/**'
+      - '.github/workflows/localisation.yml'
+      - '.editorconfig'
+      - '.gitattributes'
+      - '.gitignore'
+      - '.vscode/**'
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:18]: Nice one, I didn't know about these :)

**Comment by Gymnasiast** [comment]
This addresses too many things at once. Basically, every point you mention in the opening post should have been its own PR.

Can you create separate PRs? It will make reviewing much easier and vastly increase the speed at which the improvements can be merged.

**Comment by janisozaur** [comment]
I wanted to review this today, but got sucked into other things. I'm not sure if separate PRs are required, but that does help - or at least having it split into commits. I'm probably not going to be bothered as much, though, and if there is nothing breaking out would likely be fine with me.

**Comment by tupaschoal** [comment]
While I agree separate PRs would make it easier to merge, I think one per change might be too much. Having the different changes split by commit, though, would indeed make reviewing this a bit easier.

**Review by janisozaur — COMMENTED** [review]
Overall I quite like the changes, but would suggest splitting it into following commits:
1. `setenv` stuff (which I like the most, as it significantly cuts down on repeating the code)
2. `set -e` stuff
3. hoisting various shell `if`s to action `if`s
4. other stuff

Other than that I only have a question about rationale for replacing `wget` with `curl`, can you provide some details?


Thanks

**Comment by Margen67** [comment]
> @Margen67 any update on this?

Can I split this into smaller PRs?

**Comment by IntelOrca** [comment]
@Margen67 yep

**Comment by tupaschoal** [comment]
Is there anything left here after the merge of #16770? If so, please rebase and let us know, if not, let's just close the PR.

**Comment by Gymnasiast** [comment]
Since OP agreed with splitting this up into smaller PRs (one of which has been merged already), closing this bigger one.

**Pull request closed without merging**